doc/services/pm/device.rst

```diff
@@ -202,58 +210,319 @@ power management implementation.
 
 Use :c:macro:`PM_DEVICE_DEFINE` or :c:macro:`PM_DEVICE_DT_DEFINE` to define the
 power management resources required by a driver. These macros allocate the
-driver-specific state which is required by the power management subsystem.
+driver-specific context which is required by the power management subsystem.
 
 Drivers can use :c:macro:`PM_DEVICE_GET` or
-:c:macro:`PM_DEVICE_DT_GET` to get a pointer to this state. These
+:c:macro:`PM_DEVICE_DT_GET` to get a pointer to this context. These
 pointers should be passed to ``DEVICE_DEFINE`` or ``DEVICE_DT_DEFINE``
 to initialize the power management field in each :c:struct:`device`.
 
-Here is some example code showing how to implement device power management
-support in a device driver.
+The following example code shows how to implement device power management
+support in a device driver. Note that return values are explicitly ignored
+for brevity, in real drivers they must be handled.
 
 .. code-block:: c
 
-    #define DT_DRV_COMPAT dummy_device
+   #include <zephyr/pm/device.h>
+   #include <zephyr/pm/device_runtime.h>
 
-    static int dummy_driver_pm_action(const struct device *dev,
-                                      enum pm_device_action action)
-    {
-        switch (action) {
-        case PM_DEVICE_ACTION_SUSPEND:
-            /* suspend the device */
-            ...
-            break;
-        case PM_DEVICE_ACTION_RESUME:
-            /* resume the device */
-            ...
-            break;
-        case PM_DEVICE_ACTION_TURN_ON:
-            /*
-             * powered on the device, used when the power
-             * domain this device belongs is resumed.
-             */
-            ...
-            break;
-        case PM_DEVICE_ACTION_TURN_OFF:
-            /*
-             * power off the device, used when the power
-             * domain this device belongs is suspended.
-             */
-            ...
-            break;
-        default:
-            return -ENOTSUP;
-        }
+   #define DT_DRV_COMPAT dummy_device
 
-        return 0;
-    }
+   struct dummy_driver_data {
```

Comment:
That is a lot of code that sooner or later will be come stale , how about updating samples/subsys/pm/device_pm/, make it an annonated sample and instead reference the code from that sample here instead? This way we will know the code works and compiles and we have things in one place only.

Reply:
I've had this issue in other areas, I'd like to have compiled inline samples in the docs more or less, but the only way to do this today is make it a full fledged sample with independent docs and all of that... which makes no sense given its meant to be inline in the context of the docs.

I asked before about this before but we didn't come to any agreement on how best to do this.

Ideal scenario in my mind:
* inline doc samples go into a dedicated samples/doc_inline directory or something like it
* this directory contains samples meant to be *inlined* into docs and can therefore not be required to be documented as independent samples